jcl/src/java.base/share/classes/jdk/internal/foreign/abi/UpcallLinker.java

```diff
@@ -42,9 +42,11 @@
  * The counterpart in OpenJDK is replaced with this class that wrap up
  * an upcall handler enabling the native call to the java code at runtime.
  */
+@SuppressWarnings("nls")
 public final class UpcallLinker {
 
 	private final long thunkAddr;
+	private static final boolean isZos = System.getProperty("os.name").toLowerCase().equals("z/os");
```

Comment:
This isn't needed; instead use the preprocessor (see similar conditions in `ClassLoader.java`):
```java
		/*[IF PLATFORM-mz31 | PLATFORM-mz64]*/
		throw new InternalError("Upcall is not yet implemented"); //$NON-NLS-1$
		/*[ELSE] PLATFORM-mz31 | PLATFORM-mz64 */
		return (target, arena) -> {
			return UpcallLinker.make(target, methodType, descriptor, arena, options);
		};
		/*[ENDIF] PLATFORM-mz31 | PLATFORM-mz64 */
```